tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -229,16 +229,14 @@ def train(self, X_train, y_train, X_val=None, y_val=None, **kwargs):
     # TODO: Enable feature prune on levels > 0
     # TODO: Remove name_suffix, hacked in
     # TODO: Enable easier re-mapping of trained models -> hyperparameters input (They don't share a key since name can change)
-    def train_multi_levels(self, X_train, y_train, X_val, y_val, hyperparameters: dict = None, base_model_names: List[str] = None, hyperparameter_tune=False, feature_prune=False, core_kwargs: dict = None, level_start=0, level_end=0, stack_name_core='core', stack_name_aux='aux1', time_limit=None, name_suffix='') -> List[str]:
+    def train_multi_levels(self, X_train, y_train, X_val, y_val, X_unlabeled, hyperparameters: dict = None, base_model_names: List[str] = None, hyperparameter_tune=False, feature_prune=False, core_kwargs: dict = None, level_start=0, level_end=0, stack_name_core='core', stack_name_aux='aux1', time_limit=None, name_suffix='') -> List[str]:
```

Comment:
I would vote always making X_unlabeled an optional argument (with `None` as default value) in all the Trainer/Learner methods where you had to add it.

Reply:
Moved X_unlabeled to end of params list and set to None where ever found in abstract_trainer, auto_trainer, default_learner.